Automattic/wp-calypso
# Plugins: Move isPluginActive to the TypeScript file

## Proposed Changes

Plugin selectors should be found at `client/state/plugins/installed/selectors.js`.  However, there is one plugin selector found at `client/state/selectors/is-plugin-active`. Thus the TypeScript migration of these selectors at `client/state/plugins/installed/selectors-ts.ts` missed this selector.

This PR moves `isPluginActive` to the TypeScript file and updates it to use the new selectors so that it runs in constant time instead of searching the entire list of plugins.

## Testing Instructions



The most important thing to check is that all usages of `isPluginActive` have been migrated to the new TypeScript version.

For functional testing see the following:

**Marketplace**
Visit `/plugins/:site` and verify the following:
1. Verify that you can purchase and install a paid plugin.
2. Verify that you can install a free plugin.
3. Verify that if you visit the installation URL after the plugin is already installed and activated you are redirected to the thank you page. This URL is in this form: `/marketplace/plugin/:productSlug/install/:site`

**ECommerce Trial Upgrade Confirmation**
* On a site that has both WooCommerce and WooCommerce Payments installed and activated with WooCommerce Payments not set up, visit `/plans/my-plan/trial-upgraded/:site`.
* Click "Provide a way to pay". Verify that you are taken to `?page=wc-admin&path=%2Fpayments%2Fconnect`.
* Disable WooCommerce Payments and return to `/plans/my-plan/trial-upgraded/:site`. Click "Provide a way to pay again" and verify that you are taken to `&task=payments`, which should immediately redirect to `&path=%2Fsetup-wizard`

**Preinstalled Premium Plugins**
* Visit `/plugins/browse/paid/:site?s=jetpack+search` on a site that has Jetpack Search installed and active. Verify that the card for Jetpack Search says "Active".

**Performance & speed**
* Switch to an Atomic site.
* Make sure that the "Page Optimize" plugin is active on it.
* Go to `/settings/performance/:site`
* Verify that a "Optimize JS and CSS for faster page load and render in the browser." link appears under "Performance & speed"
* Disable the "Page Optimize" plugin and verify that the link no longer displays.

<img width="662" alt="image" src="https://github.com/Automattic/wp-calypso/assets/42627630/7514c231-adb3-4608-aeec-53ec02959be2">

## Pre-merge Checklist
